src/number.cr

```diff
@@ -239,6 +239,29 @@ struct Number
         @current += @step
       end
     end
+
+    # Overrides `Enumerable#sum` to use more performant implementation on integer
+    # ranges.
+    def sum(initial)
+      super if @reached_end
```

Comment:
Shouldn't this return?

Reply:
Haha, yeah. That's just copied 1:1 from `Range::StepIterator` 😆